<h2> Комментарий ревьюера <a class="tocSkip"></h2>

Этибар, привет!

Меня зовут Олег Михеев, и я буду проверять твой проект. Предлагаю общаться на "ты", но если тебе будет некомфортно, обязательно сообщи, и мы перейдем на "вы". Моя главная цель - не указать на твои ошибки, а поделиться опытом и подготовить тебя к работе аналитиком. Увидев у тебя ошибку, я постараюсь указать на её наличие и дам возможность найти и самостоятельно исправить её. 

В проверке проекта я буду использовать разные цвета для обозначения комментариев. Например:

<br/>

<div class="alert alert-success">
<h3> Комментарий ревьюера ✔️<a class="tocSkip"> </h3>

Так выделены удачные решения, на которые можно опираться в будущем.
</div>

<br/>

<div class="alert alert-warning">
    <h3> Комментарий ревьюера ⚠️<a class="tocSkip"> </h3>
 
Когда решение на отдельном шаге станет лучше, если внести небольшие коррективы.
</div>


<br/>
<div class="alert alert-block alert-danger">
<h3> Комментарий ревьюера ❌<a class="tocSkip"></h3>

 А так выделены решения, которые необходимо доработать, чтобы я мог принять проект.
</div>
    
  
Если вносишь изменения в проект, указывай, пожалуйста, это в своих комментариях. Будет удобно, если обозначишь их заметным цветом. Например, вот так:
       
<div class="alert alert-block alert-info">
    
<h3> Комментарий студента  <a class="tocSkip"></h3> 
    
Можешь кликнуть сюда два раза и скопировать код комментария. Также если что-то непонятно или остались вопросы по ревью, обязательно пиши.
    
</div>
    
Пожалуйста, не удаляй и не перемещай мои комментарии, чтобы не произошло путаницы, и я мог быстрее проверить проект.
<br/>

<div style="border:solid Chocolate 2px; padding: 40px">

**Общий вывод по ревью**

Хорошо, что получилось пройтись по всем шагам проекта.
    
Видно, что ты быстро усваиваешь теорию.

**Что нужно сделать, чтобы я мог принять проект**:

* Вывести сводные таблицы во всех вопросах исследовательской части и учитывать все группы клиентов при написании выводов 
* Написать подробный финальный вывод, учитывающий все группы клиентов с процентными показателями и информацию о предобработке

**Успехов! Жду проект на повторное ревью!**

<div style="border:solid Chocolate 2px; padding: 40px">

**Общий вывод по ревью v2**

Все замечания учтены.

Работа стала ещё лучше. Видно, что ты быстро усваиваешь материал и пишешь качественные выводы.

Продолжай в том же духе и станешь хорошим специалистом.

**Проект принимаю. Успехов в дальнейшем обучении!**

# Исследование надежности заемщиков


<div class="alert alert-warning">
    <h3> Комментарий ревьюера ⚠️<a class="tocSkip"> </h3>

Эта часть проверена автоматически, но я дам несколько рекомендаций, которые могут быть полезны в будущем.

Чтобы сделать проект более понятным для потенциального работодателя или коллег после названия старайся добавлять бриф, цели исследования, описание данных и план работы. Это также уменьшит количество итераций проверок во взаимодействии с другими ревьюерами в других проектах.</div>

Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [31]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

<div class="alert alert-warning">
    <h3> Комментарий ревьюера ⚠️<a class="tocSkip"> </h3>
 
 Лучше писать блоки загрузки библиотек и данных в разных ячейках. Если захочешь загрузить дополнительную библиотеку, то датасет не будет загружаться заново, что позволит избежать ошибок.
</div>

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [32]:
data.head(20)


children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [33]:
data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [34]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [35]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [36]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [37]:
data.groupby('income_type')['days_employed'].agg('median')


income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [38]:
data['children'].unique()


array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [39]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [40]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [41]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [42]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [43]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [44]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [45]:
data.duplicated().sum()

71

In [46]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [47]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [48]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [49]:
data['purpose'].unique()



array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [50]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [51]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [68]:
children_correl = data.groupby('children').agg({'debt':['count','sum']})
children_correl['debts'] = data.groupby('children')['debt'].sum()/data.groupby('children')['debt'].count() * 100
children_correl.sort_values(by='debts', ascending=False)

debt           debts
          count   sum          
children                       
4            41     4  9.756098
2          2052   194  9.454191
1          4808   444  9.234609
3           330    27  8.181818
0         14091  1063  7.543822
5             9     0  0.000000

**Вывод:** По итогам анализа видно:
1. Ранжирование размера выборки: без детей - 14 091, 1 ребенок - 4808, 2 детей - 2052, 3 детей - 330, 4 детей - 41, 5 детей - 9.
2. Соотвественно абсолютное количество должников тоже идет по ниспадающей, начиная от самой большой выборки (бездетные) до 5 детей.
4. Но! процент невозврата кредитов самый высокий среди соискателей с 4 детьми, 2мя и 1 ребенком в семье (9,8%, 9,5%, 9,2%)
Вывод: 3 группы риска невозврата долга - с 4 детьмя, 2 детьми и с 1 ребенком. Но т.к.абсолютное кол-во соискателей с 4 детьми невелико, то группы риска сокращаются до двух - с 2 детьми и с 1 ребенком. Абсолютное количество должников выше в группе с 1 ребенком.

<div class="alert alert-danger">
    <h3> Комментарий ревьюера ❌<a class="tocSkip"> </h3>
 
Если говорить только о проценте просроченных платежей, то вывод верный. Но его нельзя назвать объективным, так как не выведены данные о объёме выборок категорий заёмщиков.

Я хочу убедиться, что тебе удалось освоить сводные таблицы на хорошем уровне. Потому что в дальнейшем ты будешь пользоваться ими постоянно. Выведи, пожалуйста, общую сводную таблицу с количеством клиентов, количеством должников и процентом должников. В параметрах сводных таблиц можно указать сразу несколько агрегирующих функций.

Даю несколько ссылок, в которых есть описание, как это можно сделать:       
* С помощью pivot_table и аргумента aggfunc https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.pivot_table.html
* С помощью groupby и функции .agg() https://dfedorov.spb.ru/pandas/Подробное%20руководство%20по%20группировке%20и%20агрегированию%20с%20помощью%20pandas.html
        
При написании выводов учитывай, пожалуйста, все категории и их размеры.

</div>

<div class="alert alert-success">
<h3> Комментарий ревьюера v2 ✔️<a class="tocSkip"> </h3>

 Отлично, учтены нерепрезентативные группы. С выводом согласен.
</div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [74]:
family_correl = data.groupby('family_status').agg({'debt':['count','sum']}) #группирую в сводную таблицу с помощью метода groupby
family_correl['debts'] = data.groupby('family_status')['debt'].sum()/data.groupby('family_status')['debt'].count() * 100 #вычисляю стобец с процентом должников и добавляю его к новой таблице
family_correl.sort_values(by='debts', ascending=False)
#data['family_status'] = data['family_status'].str.lower() #привожу столбец к нижнему регистру
#family_correl = (data.groupby('family_status')['debt'].sum()/data['family_status'].value_counts()).sort_values(ascending=False) #с помощью групировки с методом sum
                                                                                         #выясняем корреляцию семейного статуса и задолженности

debt          debts
                       count  sum          
family_status                              
Не женат / не замужем   2796  273  9.763948
гражданский брак        4134  385  9.313014
женат / замужем        12261  927  7.560558
в разводе               1189   84  7.064760
вдовец / вдова           951   63  6.624606

**Вывод:** По итогам анализа видно:
1. Ранжирование размера выборки: женат/замужем - 12 261, гражданский брак - 4134, не женат/не замужем - 2796, в разводе - 1189, 
   вдовец/вдова - 951.
2. Соотвественно абсолютное количество должников тоже идет по ниспадающей, начиная от самой большой выборки (женат/замужем)
   до вдовец/вдова.
4. Но! процент невозврата кредитов самый высокий среди соискателей с семейным статусом - не женат/не замужем и в гражданском браке (9,8% и 9,3% соотвественно)
Вывод: 2 группы риска невозврата долга - неженатые/незамужние и в гражданском браке. Абсолютное количество выше в группе в гражданском браке.

<div class="alert alert-block alert-danger">
<h3> Комментарий ревьюера ❌<a class="tocSkip"></h3>

 Также выведи сводную таблицу с теми же параметрами и оцени репрезентативность групп.
</div>

<div class="alert alert-success">
<h3> Комментарий ревьюера v2 ✔️<a class="tocSkip"> </h3>

 Верно, получается, что люди, никогда не состоявшие в официальном браке, в среднем хуже выплчачивают кредиты.
</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [82]:
income_correl = data.groupby('total_income_category').agg({'debt':['count','sum']}) #группирую в сводную таблицу с помощью метода groupby
income_correl['debts'] = data.groupby('total_income_category')['debt'].sum()/data.groupby('total_income_category')['debt'].count() * 100 #вычисляю стобец с процентом должников и добавляю его к новой таблице
income_correl.sort_values(by='debts', ascending=False)
#income_correl = (data.groupby('total_income_category')['debt'].sum()/data.groupby('total_income_category')['debt'].count()).sort_values(ascending=False) #теперь группируем тем же способом по уровню дохода 

debt           debts
                       count   sum          
total_income_category                       
E                         22     2  9.090909
C                      15921  1353  8.498210
A                         25     2  8.000000
B                       5014   354  7.060231
D                        349    21  6.017192

**Вывод:** Напоминание: 
- 0–30000 — 'E';
- 30001–50000 — 'D';
- 50001–200000 — 'C';
- 200001–1000000 — 'B';
- 1000001 и выше — 'A'.
По итогам анализа видно:
1. Ранжирование размера выборки по доходам: С - 15 921, В - 5014, D - 349, А - 25, Е - 22.
2. Соотвественно абсолютное количество должников тоже идет по ниспадающей, начиная от самой большой выборки по доходам: С (50-200 тысяч) до Е (до 30 тысяч).
4. Но! процент невозврата кредитов самый высокий среди соискателей с самым низким уровнем дохода - Е (9,1%), далее с разницей в 0,5% идет самая большая группа выборки - С (8,5%).
Вывод: 2 группы риска невозврата кредитов - с доходом до 30 тысяч и с доходом от 50 до 200. Абсолютное количество должников в группе Е (до 30 тысяч) невелико - 2 человека. В группе С (от 50 до 200 тысяч) абсолютное количество в разы выше - 1353. Но стоит учитывать, что это самая многочисленная в выборке группа (15921 человек).

<div class="alert alert-block alert-danger">
<h3> Комментарий ревьюера ❌<a class="tocSkip"></h3>

 Выведи сводную таблицу по примеру из первого комментария, пожалуйста. Пересмотри вывод, если будет необходимость. Учитывай все группы клиентов.
</div>

<div class="alert alert-warning">
    <h3> Комментарий ревьюера v2 ⚠️<a class="tocSkip"> </h3>
 
Согласен, но здесь можно учитывать только группы B и C, так как они включают 98 процентов от общей выборки.
</div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [83]:
purpose_correl = data.groupby('purpose_category').agg({'debt':['count','sum']}) #группирую в сводную таблицу с помощью метода groupby
purpose_correl['debts'] = data.groupby('purpose_category')['debt'].sum()/data.groupby('purpose_category')['debt'].count() * 100 #вычисляю стобец с процентом должников и добавляю его к новой таблице
purpose_correl.sort_values(by='debts', ascending=False)

#purpose_correl = (data.groupby('purpose_category')['debt'].sum()/data.groupby('purpose_category')['debt'].count()).sort_values(ascending=False) #группируем по целям кредита

debt          debts
                          count  sum          
purpose_category                              
операции с автомобилем     4279  400  9.347978
получение образования      3988  369  9.252758
проведение свадьбы         2313  183  7.911803
операции с недвижимостью  10751  780  7.255139

**Вывод:** По итогам анализа видно:
1. Ранжирование размера выборки по целям кредита: операции с недвижимостью - 10 751, операции с автомобилем - 4279, получение образования - 3988, проведение свадьбы - 2313.
2. Соотвественно абсолютное количество должников тоже идет по ниспадающей, начиная от самой большой выборки по доходам: операции с недвижимостью до проведение свадьбы.
4. Но! процент невозврата кредитов самый высокий среди тех, чья цель это операции с автомобилем (9,4%) и получение образования (9,3%).
Вывод: 2 группы риска невозврата долгов - для операций с авто и для получения образования. Абсолютное кол-во должников в двух этих группах сопоставимо (400 и 369 соотвественно).

In [28]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              среднее             1   
8          2    6929.865299         35               высшее             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   
12         0  365286.622650         65              среднее             1   
13         0    1846.641941         54  неоконченное высшее             2   
14         0    1844.956182         56               высшее             0   
15         1     972.364419         26              среднее             1   
16         0    1719.934226         35              среднее             1   
17         0    2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0   10038.818549         48              среднее             1   

       family_status  family_status_id gender income_type  debt  total_income  \
0    женат / замужем                 0      F   сотрудник     0        253875   
1    женат / замужем                 0      F   сотрудник     0        112080   
2    женат / замужем                 0      M   сотрудник     0        145885   
3    женат / замужем                 0      M   сотрудник     0        267628   
4   гражданский брак                 1      F   пенсионер     0        158616   
5   гражданский брак                 1      M   компаньон     0        255763   
6    женат / замужем                 0      F   компаньон     0        240525   
7    женат / замужем                 0      M   сотрудник     0        135823   
8   гражданский брак                 1      F   сотрудник     0         95856   
9    женат / замужем                 0      M   сотрудник     0        144425   
10   женат / замужем                 0      M   компаньон     0        113943   
11   женат / замужем                 0      F   сотрудник     0         77069   
12  гражданский брак                 1      M   пенсионер     0        118514   
13   женат / замужем                 0      F   сотрудник     0        130458   
14  гражданский брак                 1      F   компаньон     1        165127   
15   женат / замужем                 0      F   сотрудник     0        116820   
16   женат / замужем                 0      F   сотрудник     0        289202   
17  гражданский брак                 1      M   сотрудник     0         90410   
18    вдовец / вдова                 2      F   пенсионер     0         56823   
19         в разводе                 3      F   сотрудник     0        242831   

                                   purpose total_income_category  \
0                            покупка жилья                     B   
1                  приобретение автомобиля                     C   
2                            покупка жилья                     C   
3               дополнительное образование                     B   
4                          сыграть свадьбу                     C   
5                            покупка жилья                     B   
6                        операции с жильем                     B   
7                              образование                     C   
8                    на проведение свадьбы                     C   
9   

<div class="alert alert-block alert-danger">
<h3> Комментарий ревьюера ❌<a class="tocSkip"></h3>

 Также выведи сводную таблицу и опиши все группы клиентов, пожалуйста. Пересмотри вывод, если будет необходимость.
</div>

<div class="alert alert-success">
<h3> Комментарий ревьюера v2 ✔️<a class="tocSkip"> </h3>

 Всё верно
</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Пропуски были в двух столбцах: days_employed (стаж) и total_income (доход). В первом столбце причинами могли стать отсутствие трудового стажа (студенты), неработающие люди (пенсионеры, в декрете, безработные), а во втором столбце - те, кто не заполнил данные о свохи доходах, либо не имеющие сейчас никакого дохода.

<div class="alert alert-warning">
    <h3> Комментарий ревьюера ⚠️<a class="tocSkip"> </h3>
 
Верный пример, намеренное или ненамеренное сокрытие информации - это человеческий фактор. А какую ещё причину пропусков ты можешь назвать?
</div>

**Еще одной причиной пропуска может быть техническая ошибка: либо при загрузке данных, либо при выгрузке**

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* На медианное значение не влияют выбивающиеся из общего списка числа (чересчур большие или маленькие). Поэтому выбираем медианное, как максимально близкое к истинно среднему показателю значений.

<div class="alert alert-success">
<h3> Комментарий ревьюера ✔️<a class="tocSkip"> </h3>

Абсолютно верно, такие сильно выделяющиеся значения ещё называют выбросами.
</div>

### Шаг 4: общий вывод.

Исходя из результатов проведенного анализа, есть я прихожу к выводу, что:
1. Самый высокий процент невозварта среди семей с 4 детьми. При этом семьи с 2 детьми и с 1 ребенком тоже имеют высокий процент невозврата долга.
2. Неженатые/незамужние, а также люди, живущие в гражданском браке имеют также высокий процент невозврата кредитов.
3. Люди с доходом до 30 000 рублей имеют самый высокий процент невозврата кредитов.
4. Операции с автомобилем и образование - 2 цели для кредитов, которые имеют самый выскоий процент невозврата.

**Задачей для анализа данных было узнать, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. В качестве входных данных — статистика о платёжеспособности клиентов от банка.
Результаты исследования будут учтены как рекомендация при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

Обработка данных.
1. При обработке данных были найдены пропуски в данных о трудовом стаже и уровне дохода, а также аномальные значения в данных о кол-ве детей. Пропуски в данных заполнил медианным значением по данным  виде деятельности. Таким образом каждый пропуск заполнен максимально приближенным значениям по типу деятельности.
2. Аномальные значения в данных о детях удалены, чтобы не влиять на итоги анализа.
3. Данные о доходах приведены в целочисленный формат.
4. Данные об образовании приведены к нижнему регистру и удалены дубликаты.
5. Добавлены 2 столбца в датафрейм для категоризации данных о доходах и целях кредита.

Выводы:
**Гипотеза 1. Количество детей влияет на погашение кредита в срок**
Исходя из анализа данных, я прихожу к выводу, что наличие детей влияет на погашение кредита в срок. Группа без детей самая многичсленная, но при этом с наименьшим процентом невозврата долга - 7,5% (кроме группы с 5 дтьми, которая немногочисленна сама по себе). Группы риска - соискатели с 4,2 и 1 детьми (9,8%, 9,5%, 9,2%).
**Гипотеза 2. Семейное положение влияет на погашение кредита в срок**
Исходя из анализа данных, я прихожу к выводу, что семейное положение влияет на выплату кредита в срок. Группы риска - неженатые/незамужние, а также живущие в гражданском браке (9,8% и 9,3% невозврата). Эти 2 группы многочисленны и составляют 32% от общего числа респондентов.
Также при анализе данных, было выявлено, что основные группы риска по целям на кредит это "операции с автомобилем", а также "для получения образования" - 9,3%. 
Аналитика также показала, что группа риска в размере дохода - это соискатели с доходом от 50 до 200 тысяч рублей в месяц (8,5%). Но т.к. это самая многочисленная по кол-ву респондентов группа, рекомендую не использовать для скоринга данные по доходам.

<div class="alert alert-danger">
    <h3> Комментарий ревьюера ❌<a class="tocSkip"> </h3>
    
Вывод описывает все шаги исследовательской части, но недостаточно подробно.

Твой потенциальный руководитель, в первую очередь, обратит внимание на общий вывод. Поэтому важно отразить в нём все основные результаты из всех шагов проекта. Необходимо описать информацию о проведённой предобработке, в числах описать выявленные закономерности. Сделай это, пожалуйста, учитывая все репрезентативные группы клиентов и их процентные показатели. 
 
</div>

<div class="alert alert-success">
<h3> Комментарий ревьюера v2 ✔️<a class="tocSkip"> </h3>

 Супер, вывод теперь учитывает основные группы клиентов исследовательской части и описывает все этапы проекта.
</div>